library/std/src/sys/unix/thread_local_dtor.rs

```diff
@@ -93,7 +93,7 @@ pub unsafe fn register_dtor(t: *mut u8, dtor: unsafe extern "C" fn(*mut u8)) {
     }
 }
 
-#[cfg(target_os = "vxworks")]
+#[cfg(target_os = "vxworks", target_os = "horizon")]
 pub unsafe fn register_dtor(t: *mut u8, dtor: unsafe extern "C" fn(*mut u8)) {
```

Comment:
Does this make sense to do before enabling `#[thread_local]`?

Reply:
I don't think there's any harm in it, since the only usage of this function is behind a `#[cfg(target_thread_local)]`: https://github.com/ian-h-chamberlain/rust-horizon/blob/feature/target-thread-local/library/std/src/thread/local.rs#L531

That said, I realized while double checking it that I screwed up the syntax of this `cfg`, so let me fix that...

Reply:
> Does this make sense to do before enabling `#[thread_local]`?

https://github.com/rust-lang/rust/pull/95341?notification_referrer_id=NT_kwDOA0wZd7MzMzM3NzcwNTY5OjU1MzE4OTAz#issuecomment-1079767573

Will be merged in a couple hours by bors. This PR can get merged.